azurerm/internal/services/subscription/subscription_resource.go

```diff
@@ -0,0 +1,499 @@
+					"Active",
+					"Cancelled",
+				}, false),
+			},
+
+			"billing_account": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				Description:  "The name of the billing account under which the Subscription will be created.",
+				ValidateFunc: azValidate.NoEmptyStrings, // TODO - Put some actual protection around this?
+			},
+
+			"enrollment_account": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The name of the enrollment account in which to create the subscription. Used for EA accounts.",
+				ConflictsWith: []string{
+					"invoice_section",
+					"billing_profile",
+					"subscription_id",
+				},
+				ValidateFunc: azValidate.NoEmptyStrings, // TODO - Put some actual protection around this?
+			},
+
+			"billing_profile": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The name of the Billing Profile under which the Subscription should be created. Used for MCA and Partner Agreements.",
+				ConflictsWith: []string{
+					"enrollment_account",
+					"subscription_id",
+				},
+				RequiredWith: []string{
+					"invoice_section",
+				},
+				ValidateFunc: azValidate.NoEmptyStrings, // TODO - Put some actual protection around this?
+			},
+
+			"invoice_section": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Sensitive:   true,
+				Description: "The Invoice Section of the Billing Profile which will be used for the subscription. Used for MCA and Partner Agreements.",
+				ConflictsWith: []string{
+					"enrollment_account",
+					"subscription_id",
+				},
+				RequiredWith: []string{
+					"billing_profile",
+				},
+				ValidateFunc: azValidate.NoEmptyStrings, // TODO - Put some actual protection around this?
+			},
+
+			// Optional
+			"workload": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The workload type for the Subscription. Possible values are `Production` (default) and `DevTest`.",
+				// Other RP's have updated Constants with contextual prefixes so these are likely to change
+				ValidateFunc: validation.StringInSlice([]string{
+					string(subscriptionAlias.Production),
+					string(subscriptionAlias.DevTest),
+				}, false),
+				// Workload is not exposed in any way, so must be ignored if the resource is imported.
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old == ""
+				},
+			},
+
+			"subscription_id": {
+				Type:        schema.TypeString,
+				Description: "The GUID of the Subscription.",
+				ForceNew:    true,
+				Optional:    true,
+				Computed:    true,
+			},
+
+			"tenant_id": {
+				Type:        schema.TypeString,
+				Description: "The Tenant ID to which the subscription belongs",
+				Computed:    true,
+			},
```

Comment:
we should probably also consider adding tags to these resources too, since they can be set at the subscription level

Reply:
I can't find API / SDK support for managing tags, only reading back. I'll add as `computed` for now.